### PR TITLE
fix: use icons without "size" values

### LINKED ISSUE #669: sp-icon size attribute should be deprecated
### Expected Behaviour
The sp-icon size attribute as it exists should be deprecated or removed as it doesn't map to an actual Spectrum design concept. It also defaults to the small size which isn't the default size that should be used in controls.

Instead we should make the following changes (likely with support from SpectrumCSS):

* Default the size to the equivalent of "s" in the current implementation (18x18 or 22x22 depending on the scale).
* Use an overridable variable for the default size, which will better prepare us for t-shirt sized components. E.g. we might have a large size for action button which needs the icon to be larger as well. This should "just work" if you use sp-icon. 

### Actual Behaviour
We support a size attribute that doesn't make much sense to support at this point. There are 2 upcoming spectrum initiatives that should supersede what's here: t-shirt sizing for controls & icon sizing for things with a wider range of sizes like avatars.

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Remove the use of the `size` attribute in `sp-icon` when scoped to `sp-button` and `sp-action-button` components.
- default to `size='m'` in the `:host` CSS so that it's easily overridden by the specific sizes and by scoped values of CSS Custom properties
- surface upcoming CSS Custom Property application in our local implementation until Spectrum CSS ships this update
- no usage changes required new golden images, but added icons to `sp-button` VRT for security

This doesn't "deprecate" the size attribute, as it can still be used as an override, but it's usage is lowered in priority.

Available for live review:
https://5ffc5bf40cc6c30ed2338e23--spectrum-web-components.netlify.app/storybook/index.html?path=/story/action-button-sizes--s
https://5ffc5bf40cc6c30ed2338e23--spectrum-web-components.netlify.app/storybook/index.html?path=/story/button-primary-sizes--s

## Related Issue
fixes #669

## Motivation and Context
Simplify!

## How Has This Been Tested?
VRTs passed for the `sp-action-button` et al modifications.
VRTs were added for `sp-button` with icon usage

## Types of changes
- [x] refactor

## Checklist:
- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by najikahalsema — APPROVED** [review]

File: packages/accordion/src/AccordionItem.ts [hunk 1]
```diff
@@ -97,11 +97,7 @@ export class AccordionItem extends Focusable {
                 >
                     ${this.label}
                 </button>
-                <sp-icon
-                    id="indicator"
-                    size="xs"
-                    class="spectrum-UIIcon-ChevronRight100"
-                >
+                <sp-icon id="indicator" class="spectrum-UIIcon-ChevronRight100">
```
Comment (adixon-adobe) [inline packages/accordion/src/AccordionItem.ts:100]: was this just not supposed to be `xs` before?

File: packages/action-group/stories/action-group.stories.ts [hunk 2]
```diff
@@ -322,17 +322,17 @@ export const compactIconsOnlyJustified = (): TemplateResult => {
...
             </sp-action-button>
             <sp-action-button label="Info">
-                <sp-icon slot="icon" size="m">
+                <sp-icon slot="icon">
                     ${InfoIcon({ hidden: true })}
                 </sp-icon>
             </sp-action-button>
             <sp-action-button label="View All Tags">
-                <sp-icon slot="icon" size="m">
+                <sp-icon slot="icon">
```
Comment (adixon-adobe) [inline packages/action-group/stories/action-group.stories.ts:335]: I like the changes to this file!

File: packages/button/stories/button-cta.stories.ts [hunk 3]
```diff
@@ -67,6 +67,22 @@ export const withIcon = (): TemplateResult => {
...
+        <sp-button label="Edit" size="xl" variant=${variant}>
+            <sp-icon slot="icon" size="s">
+                ${HelpIcon({ hidden: true })} Testing
+            </sp-icon>
+        </sp-button>
+        <h1>For testing purposes only</h1>
+        <p>
+            Spectrum specifies that icon elements inside of buttons should share
+            the same "size" rating or have icon accept size values as scoped in
+            the CSS by the button element.
```
Comment (adixon-adobe) [inline packages/button/stories/button-cta.stories.ts:81]: I'm not totally following what this is trying to say. I'm guessing there's a clearer way to word this.

**Review by adixon-adobe — APPROVED** [review]
I've got a couple of questions, and am a little confused by some wording in the override stories, but the actual changes look great. I don't see anything that'd block merging.

Very glad to finally see this change!

File: packages/accordion/src/AccordionItem.ts (hunk 1 above)
Comment (Westbrook) [reply packages/accordion/src/AccordionItem.ts:100]: `xs` was always a _approximation_ of the size here. The class was actually overriding it as this is a UI icon. With the update directly before this I could have used `size="none"` and be more semantically correct, but this seems like a good opportunity to flush things we don't need. What's more, searching for `<sp-icon` and removing _every_ `size` attribute seemed like a good test for this change...

File: packages/button/stories/button-cta.stories.ts (hunk 3 above)
Comment (Westbrook) [reply packages/button/stories/button-cta.stories.ts:81]: This is a test to ensure that sizing the icon will still work when it's in the scope of a parent element. You shouldn't do this on a regular basis as it is a departure from the Spectrum specification. 

Is that better, or worse, or does it give you any good ideas here?

**Pull request merged**